src/track/track.cpp

```diff
@@ -343,7 +343,12 @@ bool Track::trySetAndLockBeats(mixxx::BeatsPointer pBeats) {
 }
 
 bool Track::setBeatsWhileLocked(mixxx::BeatsPointer pBeats) {
-    if (m_pBeats == pBeats) {
+    if (m_pBeats == pBeats &&
+            // Changing the BPM of the indirectly referenced beat grid object does not
+            // immediately affect the track's nominal BPM!
+            // TODO: Using two disjunct QObject instances is a design flaw that needs
+            // to be fixed eventually.
+            m_record.getMetadata().getTrackInfo().getBpm() == getBeatsPointerBpm(pBeats)) {
```

Comment:
I don't understand this check. If the mixxx::BeatsPointer to a const beats object does not change, why this requires a change of the metadata. 
If the Metadata bpm has changed without changing the beats object. This is the wrong place to fix it.

Reply:
:thinking: I think you are right. I didn't remember that the beat grid became immutable recently. Need to check again and will close this PR.